python/ray/external_storage.py

```diff
@@ -91,6 +124,84 @@ def restore_spilled_objects(self, keys):
                 self._put_object_to_store(metadata, buf_len, f, ref)
 
 
+class ExternalStorageSmartOpenImpl(ExternalStorage):
+    """The external storage class implemented by smart_open.
+    (https://github.com/RaRe-Technologies/smart_open)
+
+    Smart open supports multiple backend with the same APIs.
+
+    Args:
+        uri(str): Storage URI used for smart open.
+        prefix(str): Prefix of objects that are stored.
+        override_transport_params(dict): Overriding the default value of
+            transport_params for smart-open library.
+
+    Raises:
+        ExternalStorageConfigInvalidError: If it fails to setup.
+            For example, if smart open library
+            is not downloaded, this will fail.
+    """
+
+    def __init__(self,
+                 uri: str,
+                 prefix: str = "ray_spilled_object_",
+                 override_transport_params: dict = None):
+        try:
+            from smart_open import open  # noqa
+        except ModuleNotFoundError:
+            raise ExternalStorageConfigInvalidError(
+                "S3 storage is chosen to be a object spilling "
+                "external storage, but smart_open "
+                "is not downloaded.")
```

Comment:
It would be better to not catch this error and just allow it to propagate. Otherwise, it is unclear what the actual cause is.